src/Appium.Net/Appium/Service/AppiumServiceBuilder.cs

```diff
@@ -450,6 +478,19 @@ private string Args
             {
                 List<string> argList = new List<string>();
                 CheckAppiumJS();
+
+                if (NodeOptions != null)
+                {
+                    foreach (var argPair in NodeOptions)
+                    {
+                        argList.Add(argPair.Key);
+                        if (!string.IsNullOrEmpty(argPair.Value))
```

Comment:
null verification is already done in the builder

Reply:
Check is still necessary since field isn't set on class creation. If you are ok with always initializing `NodeOptions` then we can ditch this null check.

Reply:
Unnecessary since last commit